src/libraries/System.Runtime/tests/System/CharTests.cs

```diff
@@ -207,6 +207,50 @@ public static void IsAscii_Char()
             Assert.False(char.IsAscii('\x0080'));
             Assert.False(char.IsAscii(char.MaxValue));
         }
+        [Fact]
+        public static void IsAsciiDigit_Char() =>
+            IsAsciiCategory(
+                char.IsAsciiDigit,
+                UnicodeCategory.DecimalDigitNumber);
+
+        [Fact]
+        public static void IsAsciiLetter_Char() =>
+            IsAsciiCategory(
+                char.IsAsciiLetter,
+                UnicodeCategory.UppercaseLetter,
+                UnicodeCategory.LowercaseLetter);
+
+        [Fact]
+        public static void IsAsciiLetterLower_Char() =>
+            IsAsciiCategory(
+                char.IsAsciiLetterLower,
+                UnicodeCategory.LowercaseLetter);
+
+        [Fact]
+        public static void IsAsciiLetterUpper_Char() =>
+            IsAsciiCategory(
+                char.IsAsciiLetterUpper,
+                UnicodeCategory.UppercaseLetter);
+
+        [Fact]
+        public static void IsAsciiLetterOrDigit_Char() =>
+            IsAsciiCategory(
+                char.IsAsciiLetterOrDigit,
+                UnicodeCategory.UppercaseLetter,
+                UnicodeCategory.LowercaseLetter,
+                UnicodeCategory.TitlecaseLetter,
+                UnicodeCategory.ModifierLetter,
+                UnicodeCategory.OtherLetter,
```

Comment:
Titlecase/Modifier/Other have no ASCII entries in the test matrix

https://github.com/dotnet/runtime/blob/6b70ca546d7a63db2cced77243dd10ce56e2c903/src/libraries
/System.Runtime/tests/System/CharTests.cs#L1072-L1104

https://www.compart.com/en/unicode/category/Lt, https://www.compart.com/en/unicode/category/Lm, and https://www.compart.com/en/unicode/category/Lo agree with that, showing that none of those categories have any members from U+0000 to U+007F (7-bit ASCII).

So... this is technically OK; but seems to imply that some things from Lt/Lm/Lo should return true, when really they all will return false.

Just thought it was worth pointing out.

Reply:
Agree with Jeremy here - ASCII does not have a concept of Titlecase / Modifier / Other letters. It's strictly upper + lower.